src/include/migraphx/onnx.hpp

```diff
@@ -58,6 +58,9 @@ struct onnx_options
     int64_t limit_max_iterations = std::numeric_limits<uint16_t>::max();
     /// Use dynamic output for operators when available
     bool use_dyn_output = false;
+    ///
+    bool is_params_cmd = false;
+    bool is_cli_cmd    = false;
```

Comment:
The onnx parser should be independent of the driver since its used for many cases. Its very hacky to add in flags like is_*_cmd, and makes the code more confusing since these flags will change the semantics of the `default_dim_value` flag to mean `batch_dimension`.

Instead of adding two flags here, you can add just one flag `batch_dimension`, which will also make the intention of the flag clearer.